src/server/routes/image-api.js

```diff
@@ -116,6 +124,86 @@ router.get('/image/:id/tags', (req, res) => {
   });
 });
 
+router.get('/image/:id/author', (req, res) => {
+    const id = req.params.id;
+    Image.findById(id, (err, image) => {
+        abortOnError(err, res);
+        res.send(image.author);
+    });
+})
+
+router.post('/image/:id/gallerythumbnail', (req,res) => {
+  const id = req.params.id;
+
+  const canWriteImage = hasRestrictions(
+      req,
+      Restrictions.WRITE_GALLERY | Restrictions.WRITE_IMAGES
+  );
+
+  if (!canWriteImage) {
+    res.status(403).end();
+    Logger.warn(`User ${req.session.user.cid} had insufficient permissions to change thumbnail.`);
+    return;
+  }
+
+  // Find the image that should be set as thumbnail
+  Image.findOne({_id: id}, (err, newThumb) => {
+    abortOnError(err, res);
+
+    // Remove the image that was previously thumbnail
+    Image.find({galleryId: newThumb.galleryId, isGalleryThumbnail: true}, (err, oldThumbs) => {
```

Comment:
imo it would make more sense to attach the thumbnail id to the gallery itself, rather than setting a boolean value on one image - with the assumption that *only one image will have isGalleryThumbnail=True*. 

Having a field on the gallery itself would ensure that each gallery would only have **one** thumbnail, and make this logic more simple :)